taranis/distance.py

```diff
@@ -0,0 +1,76 @@
+import io
+import logging
+import pandas as pd
+import subprocess
+import rich
+import sys
+from pathlib import Path
+import taranis.utils
+
+log = logging.getLogger(__name__)
+stderr = rich.console.Console(
+    stderr=True,
+    style="dim",
+    highlight=False,
+    force_terminal=taranis.utils.rich_force_colors(),
+)
+
+
+class DistanceMatrix:
+    def __init__(
+        self,
+        file_path: str,
+    ) -> "DistanceMatrix":
+        """DistanceMatrix instance creation
+
+        Args:
+            file_path (str): Locus file path
+
+        Returns:
+            DistanceMatrix: created instance
+        """
+        self.file_path = file_path
+
+    def create_matrix(self) -> pd.DataFrame:
+        """Create distance matrix using external program called mash
+
+        Returns:
+            pd.DataFrame: Triangular distance matrix as panda DataFrame
+        """
+        allele_name = Path(self.file_path).stem
+        mash_distance_command = [
+            "mash",
+            "triangle",
+            "-i",
+            self.file_path,
+            "-k",
+            "17",
```

Comment:
I'll try to include the mash parameters as parameters in case we need to adjust them

Reply:
Yes, I will set as input parameters